File: src/ui/AMP.EMS/src/app/core/models/event-guest-invitation.ts
import { EventGuest } from "./event-guest";
import { EventGuestInvitationRsvp } from "./event-guest-invitation-rsvp";

export interface EventGuestInvitation {
    eventGuestId?: string;
    eventInvitationId?: string;
    maxGuests?: number;
    code?: string;
    eventGuestInvitationRsvps?: EventGuestInvitationRsvp[];
    eventGuest?: EventGuest;
}

File: src/api/AMP.EMS.API/Core/Entities/EventGuestInvitation.cs
using AMP.Infrastructure.Entity;

namespace AMP.EMS.API.Core.Entities;

public class EventGuestInvitation : FullAuditableEntity<Guid>
{
    public Guid EventGuestId { get; set; }
    public required Guid EventInvitationId { get; set; }
    public int MaxGuests { get; set; }
    public string Code { get; set; }
    public EventGuest? EventGuest { get; set; }
    public EventInvitation? EventInvitation { get; set; }
    public ICollection<EventGuestInvitationRsvp> EventGuestInvitationRsvps { get; set; } = [];
}